File: C/Chap7_Signaux/E72/affiche.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <unistd.h>

/* handler du signal */
void handler(int sig)
{
    FILE* f = fopen("/tmp/entier.txt", "r");

    if(f == NULL)
    {
        perror("Erreur ouverture fichier");
        return;
    }

    int x;
    fscanf(f, "%d", &x);
    fclose(f);

    printf("Valeur reçue : %d\n", x);
}

int main()
{
    printf("PID de affiche : %d\n", getpid());

    signal(SIGUSR1, handler);

    /* attente infinie */
    while(1)
    {
        sleep(1);
    }

    return 0;
}

File: Python/Chap7_Signaux/P72/affiche.py
import signal
import time

def handler(sig, frame):
    try:
        with open("/tmp/entier.txt", "r") as f:
            x = f.read().strip()
            print("Valeur reçue :", x)
    except:
        print("Erreur lecture fichier")

def main():
    signal.signal(signal.SIGUSR1, handler)

    import os
    print("PID du programme affiche :", os.getpid())

    while True:
        time.sleep(1)  # attente

if __name__ == "__main__":
    main()
